frontend/device/sdl/device.lua

```diff
@@ -6,6 +6,42 @@ local logger = require("logger")
 local function yes() return true end
 local function no() return false end
 
+-- xdg-open is used on most linux systems
+local function hasXdgOpen()
+    local std_out = io.popen("xdg-open --version 2>/dev/null")
+    local version = nil
+    if std_out ~= nil then
+        version = std_out:read()
+        std_out:close()
+    end
+    return version ~= nil
+end
+
+-- open is the macOS counterpart
+local function hasMacOpen()
+    local std_out = io.popen("open")
+    local all = nil
+    if std_out ~= nil then
+        all = std_out:read()
+        std_out:close()
+    end
+    return all ~= nil
+end
+
+-- get the name of the binary used to open links
+local function getLinkOpener()
+    local enabled = false
+    local tool = nil
+    if hasXdgOpen() then
```

Comment:
Might be a bit wasteful on non-Linux when we already have `jit.os`, which will return `Linux` or `OSX`. Maybe something like this?

```suggestion
    if jit.os == "Linux" and hasXdgOpen() then
```